src/types/core.d.ts

```diff
@@ -0,0 +1,136 @@
+export as namespace core
+
+export interface ResourcePage<R> {
+  data: R[]
+  links: { [key: string]: string | null }
+  meta: {
+    page: {
+      current: number
+      limit: number
+      offset: number
+      total: number
+    }
+    results: {
+      total: number
+    }
+  }
+}
+
+export interface Resource<R> {
+  data: R
+}
+
+export interface Relationship<T> {
+  data: {
+    id: string
+    type: T
+  }
+}
+
+export namespace core {
+  export interface QueryableResource<R, F, S, I> {
+    All<ER extends R = R>(token?: string): Promise<ResourcePage<ER>>
+
+    Get<ER extends R = R>(id: string, token?: string): Promise<Resource<ER>>
+
+    Filter(filter: F): QueryableResource<R, F, S, I>
+
+    Limit(value: number): QueryableResource<R, F, S, I>
+
+    Offset(value: number): QueryableResource<R, F, S, I>
+
+    Sort(value: S): QueryableResource<R, F, S, I>
+
+    With(includes: I | I[]): QueryableResource<R, F, S, I>
+  }
+
+  export interface CrudQueryableResource<R, F, S, I> extends QueryableResource<R, F, S, I> {
+    Create<ER extends R = R>(
+      body: R
+    ): Promise<Resource<ER>>
+
+    Delete<ER extends R = R>(id: string): Promise<ResourcePage<ER>>
+
+    Update<ER extends R = R>(
+      id: string,
+      body: R
+    ): Promise<ResourcePage<ER>>
+  }
+
+  export interface RequestFactory {
+    config: Config
+    storage: StorageFactory
+
+    authenticate(): Promise<AuthenticateResponseBody>
+
+    send<T = any>(
+      uri: string,
+      method: HttpVerbs,
+      body?: any,
+      token?: string
+    ): Promise<T>
+
+    constructor(Config)
+  }
+
+  export interface ConfigOptions {
+    application?: string
+    client_id: string
+    client_secret?: string
+    currency?: string
+    host?: string
+    custom_fetch?: Function
+  }
+
+  export interface Config {
+    application?: string
+    client_id: string
+    client_secret?: string
+    host: string
+    protocol: 'https'
+    version: 'v2'
+    currency?: string
+    custom_fetch?: Function
+    auth: {
+      expires: 3600
```

Comment:
We set the expiry for this ourselves? Is 1 hour sufficient for our token expiration?

Reply:
@shaunmaharaj I'm not sure to be honest. This was just existing code I copied from moltin.d.ts - maybe worth another ticket to explore properly?

Reply:
I have actually noticed that it is a hardcoded figure in the js itself - in config.js

Reply:
@shaunmaharaj pretty sure its 1hr in the backend too

Reply:
Okay cool, we can keep it to stay consistent for now and revisit if we have to later